corehq/apps/locations/models.py

```diff
@@ -681,22 +676,11 @@ def save(self, *args, **kwargs):
             ]
             self.site_code = generate_code(self.name, all_codes)
 
-        sql_location = None
-        result = super(Location, self).save(*args, **kwargs)
-
-        # try sync locations and when SQLLocation doesn't returned, removed Couch object from database.
-        # added because when we sync location by tasks we can have behavior that the task can be
-        # killed in _sync_location method and this causes the problems
-        try:
-            sql_location = self._sync_location()
-        finally:
-            if sql_location:
-                sql_location.save()
-            else:
-                self.delete()
-                result = None
-
-        return result
+        sync_to_sql = kwargs.pop('sync_to_sql', True)
+        kwargs['sync_to_sql'] = False  # only sync here
+        super(Location, self).save(*args, **kwargs)
+        if sync_to_sql:
+            self._migration_do_sync()
```

Comment:
I saw that, but I was thinking that here both location models are already in heavy use, so a sync failure seems like it should be a hard failure.